torch/_functorch/_aot_autograd/traced_function_transforms.py

```diff
@@ -457,6 +457,27 @@ def _functionalized_f_helper(*args):
                     else:
                         inpt_old.copy_(inpt_new)
 
+            # When an output tensor is a functionalized mutated input, and we
+            # were able to move the mutation in to the graph then we can return
+            # the mutated input directly. This prevents duplicating the
+            # tensors contents.
```

Comment:
I mean, `copy_` always returns the input exactly. So in principle it's not really necessary to make this change in AOTAutograd; you could always just make the downstream Inductor pass recognize both `r = a.copy_(b); return a` and `r = a.copy_(b); return r`. I could believe that doing it this way makes the Inductor analysis involving uses/defs marginally easier, haven't read that code yet.

Reply:
The current behaviour is to return `b`. It wouldn't be valid for inductor to transform that into a return of `a` since that would introduce new aliasing between input and output tensors.